core/controllers/editor.py

```diff
@@ -775,3 +777,51 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        """Handles the GET requests for for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.PageNotFoundException
+
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
```

Comment:
We aren't handling question project in this as of now?

Reply:
Yeah currently we are doing this exploration, but later on we will extend it to questions.

Reply:
I'd say make it generic, in the sense that, the frontend can pass in any allowed entity_type (have verification for that) and if it is exploration, do the rest of it, then for questions, we can just add an 'else if' block later on.

Reply:
is this resolved? (I liked what @aks681 has suggested)

Reply:
I agree with @aks681 and @DubeySandeep. Please make sure that this is general from the get-go. 

Make sure to include an "else" clause that raises an exception or similar.

Reply:
Done

Reply:
is this resolved?

Reply:
Also I was thinking of making different get and delete controllers for exploration and question, but how should I proceed to generic way, as I am facing difficulty with ACL layer, as what will be the acl layer for generic controllers?